CLDSRV-174 update Arsenal using Armory transition work

jbertran opened the pull request:
* Use @miniscruff 's work on Armory to catch up API movements in Arsenal (CLDSRV-162).
* Fix problematic tests
* Misc fixes

This does _not_ migrate to the new error format, but uses the compatibility layer in Arsenal errors which _must_ be disabled in CLDSRV-181

Most of the fixes were merged back into the CLDSRV-162 commits 

This produces a green build in Federation - see https://eve.devsca.com/github/scality/federation/#/builders/2/builds/19435
Integration is currently DOA with setup issues, so no confirmation from there yet.

miniscruff committed:
feature: CLDSRV-162 use metadata and data from new arsenal

miniscruff committed:
feature: CLDSRV-162 update bad version ids to be proper

miniscruff committed:
feature: CLDSRV-162 use data retrieval params over fn

Commit:
CLDSRV-174 - update Arsenal

bert-e commented:
# Hello jbertran,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/cloudserver/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-174 contains:


* `7.10.4`

* `8.2.21`

* `8.3.13`

* `8.4.11`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.5`

* `8.2.21`

* `8.3.13`

* `8.4.11`

* `8.5.1`


Please check the `Fix Version/s` of CLDSRV-174, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-174 contains:


* `7.10.4`

* `8.2.21`

* `8.3.13`

* `8.4.11`

* `8.5.1`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.5`

* `8.2.21`

* `8.3.13`

* `8.4.11`

* `8.5.1`


Please check the `Fix Version/s` of CLDSRV-174, or the target
branch of this pull request.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-174 contains:


* `7.10.4`

* `8.2.21`

* `8.3.13`

* `8.4.11`

* `8.5.1`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.5`

* `8.2.21`

* `8.3.13`

* `8.4.11`

* `8.5.1`


Please check the `Fix Version/s` of CLDSRV-174, or the target
branch of this pull request.

Commit:
CLDSRV-174 - fix bucketclient error introspection

Due to circular dependency issues, we can't simply switch to a new
bucketclient. Instead, we rely on the old error format for now.

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue CLDSRV-174 contains:


* `7.10.4`

* `8.2.21`

* `8.3.13`

* `8.4.11`

* `8.5.1`



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `7.10.5`

* `8.4.11`

* `8.5.1`


Please check the `Fix Version/s` of CLDSRV-174, or the target
branch of this pull request.

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.4/feature/CLDSRV-174-use-armory-work` with contents from `feature/CLDSRV-174-use-armory-work`
and `development/8.4`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.4/feature/CLDSRV-174-use-armory-work origin/development/8.4
 $ git merge origin/feature/CLDSRV-174-use-armory-work
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.4/feature/CLDSRV-174-use-armory-work
```

bert-e commented:
# Conflict




A conflict has been raised during the creation of
integration branch `w/8.5/feature/CLDSRV-174-use-armory-work` with contents from `w/8.4/feature/CLDSRV-174-use-armory-work`
and `development/8.5`.

**I have not created the integration branch.**


Here are the steps to resolve this conflict:

```bash
 $ git fetch
 $ git checkout -B w/8.5/feature/CLDSRV-174-use-armory-work origin/development/8.5
 $ git merge origin/w/8.4/feature/CLDSRV-174-use-armory-work
 $ # <intense conflict resolution>
 $ git commit
 $ git push -u origin w/8.5/feature/CLDSRV-174-use-armory-work
```

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `feature/CLDSRV-174-use-armory-work` into
`development/7.10`
* [w/8.4/feature/CLDSRV-174-use-armory-work](https://github.com/scality/cloudserver/compare/development/8.4...w/8.4/feature/CLDSRV-174-use-armory-work) will be merged into `development/8.4` (pull request #4558)
* [w/8.5/feature/CLDSRV-174-use-armory-work](https://github.com/scality/cloudserver/compare/development/8.5...w/8.5/feature/CLDSRV-174-use-armory-work) will be merged into `development/8.5` (pull request #4559)



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`
* `stabilization/7.10.4`




*Follow* integration pull requests if you would like to be notified of
build statuses by email.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

miniscruff reviewed (approved):
I fixed the 8.4 and 8.5 branches, might want to take a quick peek to make sure though.

francoisferrand commented:
/bypass_author_approval

bert-e commented:
# Not authorized



I'm afraid I cannot do that, @francoisferrand:

> /bypass_author_approval


You don't have enough credentials to use `bypass_author_approval`.


Please **edit** or **delete** the corresponding comment so I can move on.

rahulreddy commented:
/bypass_author_approval

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.4`

* :heavy_check_mark: `development/8.5`



The following branches will **NOT** be impacted:

* `development/6.4`
* `development/7.4`
* `stabilization/7.10.4`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **bypass_author_approval**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/7.10`

* :heavy_check_mark: `development/8.4`

* :heavy_check_mark: `development/8.5`



The following branches have **NOT** changed:

* `development/6.4`
* `development/7.4`
* `stabilization/7.10.4`



Please check the status of the associated issue CLDSRV-174.

Goodbye jbertran.
